tests/unit_tests/config/test_transfer_functions.py

```diff
@@ -79,3 +80,82 @@ def test_that_truncated_normal_stretches(x, arg):
         return
     result = TransferFunction.trans_truncated_normal(x, arg)
     assert np.isclose(result, expected)
+
+
+def valid_derrf_parameters():
+    """All elements in R, min<max, and width>0"""
+    steps = st.integers(min_value=2, max_value=1000)
+    min_max = (
+        st.tuples(
+            st.floats(
+                min_value=-1e12, max_value=1e12, allow_nan=False, allow_infinity=False
+            ),
+            st.floats(
+                min_value=-1e12, max_value=1e12, allow_nan=False, allow_infinity=False
+            ),
+        )
+        .map(sorted)
+        .filter(lambda x: x[0] < x[1])  # filter out edge case of equality
+    )
+    skew = st.floats(allow_nan=False, allow_infinity=False)
+    width = st.floats(
+        min_value=0.01, max_value=1e6, allow_nan=False, allow_infinity=False
+    )
+    return min_max.flatmap(
+        lambda min_max: st.tuples(
+            steps, st.just(min_max[0]), st.just(min_max[1]), skew, width
+        )
+    )
+
+
+@given(st.floats(allow_nan=False, allow_infinity=False), valid_derrf_parameters())
+def test_that_derrf_is_within_bounds(x, arg):
+    """The result shold always be between (or equal) min and max"""
+    result = TransferFunction.trans_derrf(x, arg)
+    assert arg[1] <= result <= arg[2]
+
+
+@given(
+    st.lists(st.floats(allow_nan=False, allow_infinity=False), min_size=2),
+    valid_derrf_parameters(),
+)
+def test_that_derrf_creates_at_least_steps_or_less_distinct_values(xlist, arg):
+    """derrf cannot create more than steps distinct values"""
+    res = []
+    for x in xlist:
```

Comment:
A bit more Pythonic perhaps:

`res = [TransferFunction.trans_derrf(x, arg) for x in xlist]`

Reply:
Thanks, updated with [84d5b33](https://github.com/equinor/ert/pull/6130/commits/84d5b33a158e25a446f0f26bacbaa922460763ab)